packages/teleterm/src/ui/DocumentCluster/ClusterResources/Databases/useDatabases.ts

```diff
@@ -14,19 +14,51 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+import { AttemptStatus } from 'shared/hooks/useAsync';
+
 import { useAppContext } from 'teleterm/ui/appContextProvider';
-import { useClusterContext } from 'teleterm/ui/DocumentCluster/clusterContext';
+import { Database, ServerSideParams } from 'teleterm/services/tshd/types';
 import { routing } from 'teleterm/ui/uri';
 import { GatewayProtocol } from 'teleterm/ui/services/clusters';
 
+import { useServerSideResources } from '../useServerSideResources';
+import { useClusterContext } from '../../clusterContext';
+
+function getEmptyTableText(status: AttemptStatus) {
+  switch (status) {
+    case 'error':
+      return 'Failed to fetch databases.';
+    case '':
+      return 'Searching…';
+    case 'processing':
+      return 'Searching…';
+    case 'success':
+      return 'No databases found.';
+  }
+}
```

Comment:
Any reason this is repeated three times instead of having one function which handles all three tables?

Reply:
Each table displays slightly different text. I figured it'd be easier to just have the functions be in each table rather than extract